instrumentation/kafka-streams-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkastreams/KafkaStreamsSingletons.java

```diff
@@ -45,11 +45,11 @@
     if (!KafkaPropagation.isPropagationEnabled()) {
       return builder.newInstrumenter(SpanKindExtractor.alwaysConsumer());
     } else if (ExperimentalConfig.get().suppressMessagingReceiveSpans()) {
-      return builder.newConsumerInstrumenter(new KafkaHeadersGetter());
+      return builder.newConsumerInstrumenter(new KafkaConsumerRecordGetter());
     } else {
       builder.addSpanLinksExtractor(
           SpanLinksExtractor.fromUpstreamRequest(
-              GlobalOpenTelemetry.getPropagators(), new KafkaHeadersGetter()));
+              GlobalOpenTelemetry.getPropagators(), new KafkaConsumerRecordGetter()));
       return builder.newInstrumenter(SpanKindExtractor.alwaysConsumer());
     }
```

Comment:
this whole method can be replaced now by `KafkaUtils.buildConsumerProcessInstrumenter(INSTRUMENTATION_NAME)` 👍